### ISSUE TITLE
Monkey menu text sometimes cut off at bottom

### ISSUE THREAD

**Issue by stonecrusher** [greasemonkey/greasemonkey#2868]
*Moved from other [issue](https://github.com/greasemonkey/greasemonkey/issues/2740#issuecomment-367848276)*
![screenshot 3](https://user-images.githubusercontent.com/1388389/36568358-f683cf12-1829-11e8-9cce-ebfe09f30b4f.png)
For the cut off text, as written above:
`Greasemonkey 4.3beta5`
`Firefox 60.0a1 (2018-02-22) (64-Bit)` (happens with all three default schemes)
`Windows 8.1 home`, 125% scaling (should be also pretty standard because with 100% the text is *really* small. Can't remember having changed that.)
German localization for Windows and Firefox

Does not happen with Windows 7 home x32 (english), basic and classic colour schemes, 125% scaling (I think that's standard for nowadays resolutions and called "medium"), same GM and FF versions as above, just x32.

If you need further information, just ask back, I don't know what else could be helpful. But everything should be pretty standard because those are virtual machines with barely any software on it. OS is up to date.

I can't inspect the menu by myself, I always just get to the outer container `browser.webextension-popup-browser`.
Note that it seems that only the last item in a line doesn't have proper vertical alignment. So the text of scripts in the main menu are ok, just the corresponding rightarrow is shifted down. Accordingly new script and help item texts are shifted.
/edit
another screenshot about the last note
![screenshot 3](https://user-images.githubusercontent.com/1388389/36574390-eaf8b18c-1845-11e8-97f3-ed3902ae905c.png)

**Comment by Eselce** [comment]
#2833

**Comment by arantius** [comment]
Happens: Windows 8.1; doesn't happen: Windows 7.  This is ~impossible for me to test.  I have Win7 but not Win8 anywhere.

Where I am now (on Linux):

![untitled](https://user-images.githubusercontent.com/84729/36599252-d39e221a-187c-11e8-8946-c3bc85d4aaf8.png)

The text is 16px tall and exactly the same size as/aligned vertically with the icon.  Your text seems 
like it might be bigger, and is vertically offset.

https://wiki.greasespot.net/Troubleshooting_(Users)

Does this happen in a fresh profile?

**Comment by Eselce** [comment]
Example:

Correctly aligned (list of 3 test scripts):

```
<div rv-hide="userScripts.inactive | empty">
      <hr>
      <div class="subview-item heading" rv-hide="userScripts.active | empty" style="display: none;">
        <div class="text">Sonstige Benutzerskripte</div>
      </div>

      <!-- rivets: each-script --><a class="subview-item next-menu user-script" rv-href="script.uuid | mmUuidMenu" rv-data-uuid="script.uuid" rv-class-disabled="script.enabled | not" href="#4e4b7955-01a5-47fc-a9a4-0d8d0cd903a8" data-uuid="4e4b7955-01a5-47fc-a9a4-0d8d0cd903a8">
        <div class="icon"><img rv-src="script.icon" src="moz-extension://38f8bf13-ad38-4e21-850d-514be48e0341/skin/userscript.png"></div>
        <div class="text">i18n</div>
      </a><a class="subview-item next-menu user-script" rv-href="script.uuid | mmUuidMenu" rv-data-uuid="script.uuid" rv-class-disabled="script.enabled | not" href="#4735382d-b7aa-408b-9367-9ccefe470b7c" data-uuid="4735382d-b7aa-408b-9367-9ccefe470b7c">
        <div class="icon"><img rv-src="script.icon" src="moz-extension://38f8bf13-ad38-4e21-850d-514be48e0341/skin/userscript.png"></div>
        <div class="text">i18n test</div>
      </a><a class="subview-item next-menu user-script" rv-href="script.uuid | mmUuidMenu" rv-data-uuid="script.uuid" rv-class-disabled="script.enabled | not" href="#6fcad8f6-4cd4-4497-8898-3194539793c3" data-uuid="6fcad8f6-4cd4-4497-8898-3194539793c3">
        <div class="icon"><img rv-src="script.icon" src="moz-extension://38f8bf13-ad38-4e21-850d-514be48e0341/skin/userscript.png"></div>
        <div class="text">OS2.jugend</div>
      </a>
    </div>
```

Misaligned ("New user script..."):
```
<a href="#new-user-script" class="subview-item">
      <div class="icon fa fa-file-text-o"></div>
      <div class="text">Neues Benutzerskript...</div>
    </a>
```

**Comment by Sxderp** [comment]
@Eselce , if you stick an empty `<img>` on the misaligned bits, do they align?

**Comment by Eselce** [comment]
@Sxderp good idea, will try to edit the file

The output looks the same as above, except that the text is lowered in the box, and it's cut, where the blue area ends.
For me, the blue area is 1 or more pixels beyond the dashed line. And the font is not as bold as under Linux...

**Comment by Eselce** [comment]
Well, that's interesting: I've added
` <img alt="TEXT" src="">`
which results in the text `TEXT`. This text is misaligned even more, so that the E is cut right below the middle.
The middle bar of the `t` is on the same line as the top bar of the `T` in `Aktiviert TEXT`.

I'd like to test it with other browsers, but IE sucks and on this computer, I have no other browser than Firefox (and IE).

**Comment by Eselce** [comment]
Hmm... Looks like if I substitute the icon
```
<div class="icon fa fa-..."></div>
```
e.g.
```
<div class="icon fa fa-pencil-square-o"></div>
```
with
```
<div class="icon"><img rv-src="script.icon"></div>
```
then it works fine...

**Comment by Eselce** [comment]
Although the `TEXT` in
```
<img alt="TEXT" src="">
```
is still lowered (but only once this time: same relation to `Aktiviert`, but that text is well aligned now).

**Comment by stonecrusher** [comment]
> Does this happen in a fresh profile?
yes.

I still wonder why I can't inspect the menu, but only the surrounding container. Must be doing something wrong.

**Comment by Eselce** [comment]
You could try to open the menu in a new window. Maybe that's easier to inspect...

**Comment by stonecrusher** [comment]
Search for `Benutzerskript` doesn't find anything, `webextensions-popup-stack` has no child elements.
![screenshot 3](https://user-images.githubusercontent.com/1388389/36631668-98bec5f4-197c-11e8-9e36-be70cde2d2a2.png)
Also searched through those:
![screenshot 4](https://user-images.githubusercontent.com/1388389/36631713-37de8520-197d-11e8-825a-8ab9f87cc0e4.png)
No results.

Sorry for posting this here, I couldn't figure out how to PN @Eselce

**Comment by Sxderp** [comment]
Inspect it by going to `about:debugging` and select Greasemonkey. The popup window should appear in the list then.

**Comment by stonecrusher** [comment]
Thanks @Sxderp that works.

So setting `line-height:1em !important` on all the shifted elements fixes it for me, but it's just a workaround. The culprit is the `.subview-item::after { float:right; }`.

If you remove it, you have to redesign the script section, e.g. `.subview-item .icon { display:inline-block }`

**Comment by Sxderp** [comment]
> So setting line-height:1em !important on all the shifted elements fixes it for me ...

Hm, line height is never explicitly set in `monkey-menu.css`. It could be that whatever system default Windows 8 has is screwy.

> The culprit is the `.subview-item::after { float:right; }`. Floating and -moz-box don't work well together.

Source? The Firefox UI uses `-moz-box` with a `::after` to display the arrows and keyboard shortcuts in the hamburger menu.

**Comment by stonecrusher** [comment]
Sorry, that was just a guess. Floating can cause quite strange behaviour in my experience. I use flexbox quite often by now and never came across a combination with floating, which is a bit the table layout of today.

As it works in the native FF menus, there may be another problem, I'm investigating. But it's conspicious that the rest is normal when disabling the floating.

For now a `.subview-item { line-height:1em; }` should be sufficient.
Note: The text of the script section is a bit too high then, but at least there are no cut off `p` `g` or `y`
Already played around with `self-align` and similar, but there is still the padding and everything is connected.

**Comment by Eselce** [comment]
Hmm, if I add that, it's almost okay, but for me, `p`, `g` and `y` in `Benutzerskript(e)`, `Greasemonkey` and `Homepage` are still cut off below the baseline...

**Comment by Eselce** [comment]
... and I still don't understand, why this
```
<img alt="TEXT" src="">
```
`TEXT` is lowered, compared to the normal text...

**Comment by stonecrusher** [comment]
> it's almost okay, but for me, p, g and y in Benutzerskript(e), Greasemonkey and Homepage are still cut off below the baseline...

Disable / overwrite the `.subview-item .text { overflow-x:hidden }` with `visible !important` or `unset`

> and I still don't understand

The problem is that we don't have an easy fiddle to test it. There are loads of properties applied and the box model itself is not as consistent as it should be. Btw I think `-moz-box` will disappear in future versions and replaced by flex items, which will bring more consistency.

**Comment by Eselce** [comment]
Yes, `unset` fixes that, though I think, the text is still too low. Lower than the icons...

**Comment by Sxderp** [comment]
`-moz-box` provides some convenient styling over flex. It should probably be updated but I took the shortcut route and pretty much copy pasted panel styling  

Getting flex to behave like `-moz-box` isn't particularly easy. 

How does setting or marking the overflow visible effect long script names?

**Comment by Eselce** [comment]
> How does setting or marking the overflow visible effect long script names?

The arrow `>` is gone, that's all...

**Comment by Sxderp** [comment]
That's probably because the text pushes it off to the right.

**Comment by Eselce** [comment]
Yes. I think, it looks reasonable...

**Comment by Sxderp** [comment]
I pushed a new branch. Can you test it out?
https://github.com/Sxderp/greasemonkey/tree/monkey-menu-style-box

**Comment by Eselce** [comment]
Just did it. For me, it's visually sufficient!

**Comment by Eselce** [comment]
![](file:///D:/Transfer/GM-menu-style-box.png)

**Comment by Sxderp** [comment]
> Well, the texts seem to be lower than the icons...

Could you provide an image? For me they're 'roughly centered'. Maybe like 1px off.

**Comment by Eselce** [comment]
Already had an image, tried to simplify the upload, but that failed! :frowning_face: 

![](https://github.com/Eselce/GitTest/blob/master/misc/img/GM-menu-style-box.png)

**Comment by Eselce** [comment]
![](http://eselce.github.io/GitTest/misc/img/GM-menu-style-box.png)

**Comment by Eselce** [comment]
You can see, that the baseline of the script names is now below the script icons...

**Comment by Sxderp** [comment]
RIP Windows 8 so broken. But at least it's not cut. I'll submit a pull.

**Comment by Eselce** [comment]
Yeah, great, it's really sufficient (for me)!

**Comment by stonecrusher** [comment]
Don't know what you did, but as I wrote `line-height:1em` cares about vertical position and `overflow-x:visible` about the still left cutoff of pgyq.
sorry if I missed something, I just got up.
however, if it works now, it works!

**Comment by arantius** [comment]
Because this is a confusing/subtle issue and thus hard to fix, I'm going to declare this "fixed" by #2872.  That will be included in version 4.3, and I will welcome further detailed reports against 4.3, if things aren't truly perfectly fixed yet.
